src/core/iomgr/socket_utils_common_posix.c

```diff
@@ -89,6 +89,26 @@ int grpc_set_socket_no_sigpipe_if_possible(int fd) {
 #endif
 }
 
+int grpc_set_socket_ip_pktinfo_if_possible(int fd) {
+#ifdef GPR_HAVE_IP_PKTINFO
+  int get_local_ip = 1;
+  return 0 == setsockopt(fd, IPPROTO_IP, IP_PKTINFO, &get_local_ip,
+                         sizeof(get_local_ip));
+#else
+  return 1;
```

Comment:
Please add `(void)fd` to tell some older compilers that the fact the `fd` variable isn't being referenced from that codepath isn't a problem.